src/delimited-stream.ts

```diff
@@ -0,0 +1,30 @@
+import StreamSearch = require('streamsearch');
+import { PromiseNext } from './promise-next';
+
+export type ReadCallback = (chunk: ArrayBuffer | SharedArrayBuffer) => any;
+
+export interface Readable {
```

Comment:
I suggest using `import { Readable } from 'stream';` and document that this only supports `Buffer` streams (a stream which decodes into strings is probably lossy?), and for Node.JS APIs we can rely on always being `Buffer` and never `TypedArray` or an object stream.

Then the unittests might fake using `memorystream` instead of an `EventEmitter` and the streaming can be `inputStream.pipe(searcher)` which also handles the `end` (even though that is the shell exit anyway).

Reply:
Not done: I don't think `memorystream` will let me do what I want.  Specifically, I am unsure that `memorystream` will break up events into chunks the way I'd want it to.  It did do it when I played around with it, but the documentation isn't entirely clear about _how_ it breaks things up.
OTOH, `Readable` is a pretty big interface.  Not doing this.